R/readS4Nproject_input.R

```diff
@@ -1,90 +1,93 @@
-  
+  data$Region[data$Region == "Russia+"]         <- "Russia +"
+  data$Region[data$Region == "China+"]          <- "China +"
+  data$Region[data$Region == "Rest C. America"] <- "Rest Centr America"
+  data$Region[data$Region == "Rest S. America"] <- "Rest South America"
+  data$Region[data$Region == "Rest S. Asia"]    <- "Rest South Asia"
+  data$Region[data$Region == "Rest S. Africa"]  <- "Rest Southern Africa"
+
   data      <- merge(data, regionmapping, "Region")
-  data      <- data.frame(scenario=data$Scenario,year=data$Year,value=data$Value,iso=data$iso, stringsAsFactors=F)
-  data$year <- paste0("y",data$year)
+  data      <- data.frame(scenario = data$Scenario,
+                          year = data$Year,
+                          value = data$Value,
+                          iso = data$iso,
+                          stringsAsFactors = FALSE)
+  data$year <- paste0("y", data$year)
   x         <- collapseNames(as.magpie(data))
-  
-  if (subtype=="bioenergy"){
+
+  if (subtype == "bioenergy") {
     # Disaggregate regional bioenergy to country values using population weight
     # country population in SSP2
-    pop     <- calcOutput("Population", 
-                          naming = "indicator.scenario",
-                          extension2150 = "none",
-                          aggregate = FALSE, 
-                          years = paste0("y", c(2005, seq(2010, 2100, by = 10))), 
-                          round = 6)
-    pop     <- collapseNames(pop[,,"SSP2"])
+    pop     <- collapseNames(calcOutput("Population",
+                                        scenario = "SSP2",
+                                        extension2150 = "none",
+                                        aggregate = FALSE,
+                                        years = paste0("y", c(2005, seq(2010, 2100, by = 10))),
+                                        round = 6))
 
     # regional population
-    pop_reg <- toolAggregate(pop, rel=regionmapping, weight=NULL)
-    pop_reg <- as.data.frame(pop_reg)
-    pop_reg <- data.frame(Region=pop_reg$Region, year=as.character(pop_reg$Year), Value=pop_reg$Value, stringsAsFactors=F)
-    pop_reg <- merge(pop_reg, regionmapping, "Region") 
-    pop_reg <- collapseNames(as.magpie(data.frame(iso=pop_reg$iso, year=pop_reg$year, Value=pop_reg$Value)))
+    popReg <- toolAggregate(pop, rel = regionmapping, weight = NULL)
+    popReg <- as.data.frame(popReg)
+    popReg <- data.frame(Region = popReg$Region,
+                         year = as.character(popReg$Year),
+                         Value = popReg$Value,
+                         stringsAsFactors = FALSE)
+    popReg <- merge(popReg, regionmapping, "Region")
+    popReg <- collapseNames(as.magpie(data.frame(iso = popReg$iso, year = popReg$year, Value = popReg$Value)))
 
-    # country population share
-    pop_shr <- pop/pop_reg
-    
     # bioenergy demand weighted by population
-    x <- x * pop/pop_reg
+    x <- x * pop / popReg
```

Comment:
Just a comment, but can't one simply do ` x <- toolAggregate(x, rel = regionmapping, from = "Region", to = "iso", weight = pop)` to disaggregate x?
(I didn't test this!)

Reply:
Seems reasonable, but I didn't want to change more than necessary (to fix the warning + linter), so I don't need to test more than necessary :wink: